File: test/src/cpp/t04_exists_only.cpp
#include <unistd.h>
#include <sys/stat.h>
#include <sys/utsname.h>
#include <time.h>
#include <sched.h>

class ExistsOnly {
public:
    __attribute__((used, noinline))
    static void hidden() {
        (void)::getuid();
        (void)::geteuid();
        (void)::getgid();
        (void)::getegid();
        (void)::umask(022);
    }
};

class MainFlow {
public:
    static int run() {
        utsname u{};
        timespec ts{};
        (void)::getpid();
        (void)::getppid();
        (void)::uname(&u);
        ts.tv_sec = 0;
        ts.tv_nsec = 13 * 1000 * 1000;
        (void)::nanosleep(&ts, nullptr);
        (void)::sched_yield();
        return 0;
    }
};

int main() {
    return MainFlow::run();
}

File: test/src/go/t04_exists_only.go
package main

import (
	"os"
	"runtime"
	"syscall"
	"time"
)

func existsOnly() {
	_ = os.Getuid()
	_ = os.Geteuid()
	_ = os.Getgid()
	_ = os.Getegid()
	_ = syscall.Umask(022)
}

var sink = []func(){existsOnly}

func main() {
	_ = os.Getpid()
	_ = os.Getppid()
	var u syscall.Utsname
	_ = syscall.Uname(&u)
	time.Sleep(13 * time.Millisecond)
	runtime.Gosched()

	_ = sink[0]
}